nari/ext/act/utils.py

```diff
@@ -1,44 +1,39 @@
 """Helpful utilities"""
-from enum import Enum
 from datetime import datetime
-from hashlib import md5, sha256
+from hashlib import sha256
+from typing import Callable
 
 from nari.types import Timestamp
 
-from nari.ext.act.exceptions import InvalidActChecksumAlgorithm
-
 
 DEFAULT_DATE_FORMAT: str = '%Y-%m-%dT%H:%M:%S.%f%z'
 
 
-# pylint: disable=invalid-name
-class ActLogChecksumType(Enum):
-    """List of hashsum algorithms used by different ACT versions"""
-    MD5 = "md5"
-    SHA256 = "sha256"
-# pylint: enable=invalid-name
+try:
+    from nari_act_rust.utils import date_from_cs_string, validate_checksum_internal as validate_checksum_internal_rs
+except ImportError:
+    def date_from_cs_string(datestr: str) -> Timestamp:
+        """Look, this is dirty. This is wrong. Please someone find a better way to do this."""
+        return int(datetime.strptime(f'{datestr[:26]}{datestr[-6:]}', DEFAULT_DATE_FORMAT).timestamp() * 1000)
 
 
-def date_from_act_timestamp(datestr: str) -> Timestamp:
-    """Parse timestamp from ACT log into a Timestamp
-    Look, this is dirty. This is wrong. Please someone find a better way to do this.
-    """
-    return int(datetime.strptime(f'{datestr[:26]}{datestr[-6:]}', DEFAULT_DATE_FORMAT).timestamp() * 1000)
+def validate_checksum_internal_py(line: str, index: int) -> bool:
+    """Validates an ACT log line internal function"""
+    parts = line.split('|')
+    check_hash = parts[-1]
+    to_hash = f'{"|".join(parts[:-1])}|{index}'.encode('utf-8')
+
+    return sha256(to_hash).hexdigest()[:16] == check_hash
 
+validate_checksum_internal: Callable[[str, int], bool] = validate_checksum_internal_rs or validate_checksum_internal_py
 
-def validate_checksum(line: str, index: int, algo: ActLogChecksumType = ActLogChecksumType.SHA256) -> bool:
+def date_from_act_timestamp(datestr: str) -> Timestamp:
+    """Parse timestamp from ACT log into a Timestamp"""
+    return date_from_cs_string(datestr)
+
+def validate_checksum(line: str, index: int) -> bool:
     """Validates an ACT log line
     Given some line 1|foo|bar|baz|a823425f532c540667195f641dd3649b, and an index of 1, then the md5sum of
     1|foo|bar|baz|1 (where 1 is the index) should be a823425f532c540667195f641dd3649b (which is the checksum value)
     """
-    parts = line.split('|')
-    check_hash = parts[-1]
-    to_hash = f'{"|".join(parts[:-1])}|{index}'.encode('utf-8')
-
-    match algo:
-        case ActLogChecksumType.MD5:
-            return md5(to_hash).hexdigest() == check_hash
-        case ActLogChecksumType.SHA256:
-            return sha256(to_hash).hexdigest()[:16] == check_hash
-        case _:
-            raise InvalidActChecksumAlgorithm(f'Unexpected checksum algorithm: {algo}. Expected one of MD5 and SHA256.')
+    return validate_checksum_internal(line, index)
```

Comment:
I don't like this indirection inside the function. Why can't we do the same `from nari_act_rust import foo as foo_rs` here and overwrite the whole `validate_checksum` stub here?

Reply:
Would it export from the module properly then with doing `name = name_rs or name_py`